src/particles/pusher/BeamParticleAdvance.cpp

```diff
@@ -186,19 +209,75 @@ AdvanceBeamParticlesSlice (
                     + dt * 0.5_rt * charge_mass_ratio * Ezp;
 
                 const amrex::ParticleReal gamma_intermediate_inv = 1._rt / std::sqrt( 1._rt
-                    + ux_intermediate*ux_intermediate*clightsq
-                    + uy_intermediate*uy_intermediate*clightsq
-                    + uz_intermediate*uz_intermediate*clightsq );
+                    + (ux_intermediate*ux_intermediate
+                    +  uy_intermediate*uy_intermediate
+                    +  uz_intermediate*uz_intermediate)*inv_c2 );
 
-                const amrex::ParticleReal uz_next = uz + dt * charge_mass_ratio
+                amrex::ParticleReal uz_next = uz + dt * charge_mass_ratio
                     * ( Ezp + ( ux_intermediate * Byp - uy_intermediate * Bxp )
                     * gamma_intermediate_inv );
 
+                if (radiation_reaction) {
+
+                    amrex::ParticleReal Exp = ExmByp + clight*Byp;
+                    amrex::ParticleReal Eyp = EypBxp - clight*Bxp;
+
+                    // convert to SI units, no backwards conversion as not used after RR calculation
+                    if (normalized_units) {
+                        Exp *= E0;
+                        Eyp *= E0;
+                        Ezp *= E0;
+                        Bxp *= E0;
+                        Byp *= E0;
+                        Bzp *= E0;
+                    }
+                    const amrex::ParticleReal gamma_intermediate = std::sqrt( 1._rt
+                        + (ux_intermediate*ux_intermediate
+                        +  uy_intermediate*uy_intermediate
+                        +  uz_intermediate*uz_intermediate)*inv_c2 );
```

Comment:
The indentation is confusing here, below is one way to fix it.
```suggestion
                    const amrex::ParticleReal gamma_intermediate = std::sqrt(
                        1._rt + ( ux_intermediate*ux_intermediate
                                 +uy_intermediate*uy_intermediate
                                 +uz_intermediate*uz_intermediate )*inv_c2 );
```